output/cloud/expv2/metrics_client.go

```diff
@@ -18,47 +18,39 @@ import (
 	"go.k6.io/k6/output/cloud/expv2/pbcloud"
 )
 
-type httpDoer interface {
-	Do(*http.Request) (*http.Response, error)
-}
-
 // metricsClient is a Protobuf over HTTP client for sending
 // the collected metrics from the Cloud output
 // to the remote service.
 type metricsClient struct {
-	httpClient httpDoer
+	httpClient *cloudapi.Client
```

Comment:
These are changes from #3104, so can you rebase on `cloud-v2-retries` so that we can see a clear diff here?